cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js

```diff
@@ -0,0 +1,300 @@
+        });
+    });
+
+    describe('removeCrosslisting', function() {
+        var deleteCrosslistingDeferred = null;
+
+        beforeEach(function () {
+            deleteCrosslistingDeferred = $q.defer();
+            scope.dtInstance = jasmine.createSpyObj('dtInstance', ['reloadData']);
+            spyOn(scope, 'deleteCrosslisting').and
+                .returnValue(deleteCrosslistingDeferred.promise);
+            spyOn(scope, 'handleAjaxErrorResponse');
+        });
+
+        it('calls deleteCrosslisting with the correct id', function() {
+            scope.removeCrosslisting(1, 2, 3);
+            expect(scope.deleteCrosslisting).toHaveBeenCalledWith(1);
+        });
+
+        it('shows failure message and cleans up on failure', function() {
+            scope.removeCrosslisting(1, 1, 2);
+
+            expect(scope.operationInProgress).toBe('remove');
+            expect(scope.message).toBe(null);
+
+            deleteCrosslistingDeferred.reject('failure!');
+            scope.$digest();
+
+            expect(scope.operationInProgress).toBeNull();
+            expect(scope.dtInstance.reloadData).toHaveBeenCalled();
+            expect(scope.handleAjaxErrorResponse).toHaveBeenCalled();
+            expect(scope.message.alertType).toBe('danger');
+        });
+
+        it('shows success message and cleans up on success', function() {
+            scope.removeCrosslisting(1, 1, 2);
+
+            expect(scope.operationInProgress).toBe('remove');
+            expect(scope.message).toBe(null);
+
+            deleteCrosslistingDeferred.resolve('success!');
+            scope.$digest();
+
+            expect(scope.operationInProgress).toBeNull();
+            expect(scope.dtInstance.reloadData).toHaveBeenCalled();
+            expect(scope.handleAjaxErrorResponse).not.toHaveBeenCalled();
+            expect(scope.message.alertType).toBe('success');
+        });
+
+    });
+
+    describe('submitAddCrosslisting', function() {
+        var primary = '124',
+            secondary = '456',
+            expectedPostParams = {
+                    primary_course_instance: primary,
+                    secondary_course_instance: secondary
+            },
+            expectedResponse = {
+                    status: 201  // format is unimportant for purposes of test
+            },
+            expectedFailureResponse = {
+                alertType: 'danger',
+                text: primary+' could not be crosslisted with '+secondary+' at this time. Please check the course instance IDs and try again.'
+            };
+        beforeEach(function () {
+            scope.rawFormInput.primary = primary;
+            scope.rawFormInput.secondary = secondary;
+            scope.dtInstance = {reloadData: function(){}};
+            scope.submitAddCrosslisting();
+        });
+
+        it('should make sure the postNewCrosslisting is called with the correct values', function(){
+            $httpBackend.expectPOST(xlistURL, expectedPostParams)
```

Comment:
We're actually testing postNewCrosslisting here. To isolate this test, we would want to mock it like we do for removeCrosslisting().